external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/JsonUtils.scala

```diff
@@ -68,20 +68,35 @@ private object JsonUtils {
           partOffsets.map { case (part, offset) =>
               new TopicPartition(topic, part) -> offset
           }
-      }.toMap
+      }
     } catch {
       case NonFatal(x) =>
         throw new IllegalArgumentException(
           s"""Expected e.g. {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}, got $str""")
     }
   }
 
+  def partitionTimestamps(str: String): Map[TopicPartition, Long] = {
+    try {
+      Serialization.read[Map[String, Map[Int, Long]]](str).flatMap { case (topic, partTimestamps) =>
+        partTimestamps.map { case (part, timestamp) =>
+          new TopicPartition(topic, part) -> timestamp
+        }
+      }
+    } catch {
+      case NonFatal(x) =>
+        throw new IllegalArgumentException(
+          s"""Expected e.g. {"topicA": {"0": 1549597128110, "1": 1549597128110},
```

Comment:
Why the hard-coded values here? can you make it clear this just suggests a general pattern that the input does not match?

Reply:
Hmm... that's consistent exception message with `partitionOffsets`, as it provides hard-code values to represent example of input. (These values just look oddly long, as they're representing timestamp.) Do you have suggestion to make it better?

Reply:
I think people may read it as the actual value their app should have supplied. Maybe as simple as replacing the concrete timestamps with something obviously fake like 123456789 or `[timetsamp1]`, and showing an ellipsis to indicate there could be more, etc

Reply:
Great suggestion! I'll take fake number as it's simpler.